docs/decisions/0010-limit-modulestore-use-in-lms.rst

```diff
@@ -0,0 +1,273 @@
+Limit LMS Modulestore access to the courseware app
+--------------------------------------------------
+
+Context
+=======
+
+Django apps deployed in the LMS sometimes want to make queries about course team
+authored content, like sequences, units, and problems. To do this, they often
+use the Modulestore, which can return parts of the course as a graph of XBlocks.
+Unfortunately, the Modulestore is a large, complex system that is shared by many
+features. This has been the cause of many bugs and performance issues over the
+years. Modulestore performance has been gradually improved over the years, but
+at the cost of adding even more complexity.
+
+Decisions
+=========
+
+1. New features will be implemented without making calls to the Modulestore in
+   the LMS.
+2. Teams should opportunistically remove LMS Modulestore dependencies in
+   existing features as they are being worked on.
```

Comment:
Does this include indirect dependencies? As I understand it, calling `has_access` with a usage key dives into the modulestore.

Do we want to warn against using that or any other utility functions that are big modulestore offenders? Or are you hoping just to focus on direct dependencies here, and tackle utilities later?

Reply:
Yeah, I'm hoping to hit the main apps and then deal with the extended permissions stuff elsewhere as a separate effort. I think it's a much more straightforward ask to ask people to remove modulestore usage from the things their team owns, while asking them to remove access checks from the bowels of `has_access` is a much harder sell. We will have to cross that bridge (and burn it to cinders) at some point though.